src/System.Management.Automation/engine/Utils.cs

```diff
@@ -478,13 +478,22 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
         }
 #endif
 
-        internal static string DefaultPowerShellAppBase => GetApplicationBase(DefaultPowerShellShellID);
+        internal static string DefaultPowerShellAppBase => GetApplicationBase();
```

Comment:
The `AddType.cs` needs to be updated too:
https://github.com/PowerShell/PowerShell/blob/8f0cd65257ab8959dd16a424a5973fbf2fdb7651/src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs#L634-L640

`s_netcoreAppRefFolder` should be just using `GetApplicationBase()`. As for `s_frameworkFolder`, the reference of this variable can be replaced by `GetApplicationBase()` directly I think.